proposals/NNNN-Vehicle-data-refactoring.md

```diff
@@ -0,0 +1,598 @@
+<struct name="TurnSignalData" since="X.x">
+	<param name="turnSignal" type="TurnSignal" mandatory="false" since="5.0">
+		<description>See TurnSignal</description>
+	</param>	
+	<param name="hazardLights" type="VehicleDataStatus" mandatory="false" since="X.x">
+		<description>Status of hazard lights</description>
+	</param>
+</struct>
+```
+
+##### Add to function `SubscribeVehicleData` request: 
+
+```xml
+<param name="turnSignal" type="Boolean" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+##### Add to function `SubscribeVehicleData` response: 
+
+```xml
+<param name="turnSignal" type="VehicleDataResult" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+##### Add to function `UnsubscribeVehicleData` request: 
+
+```xml
+<param name="turnSignal" type="Boolean" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+##### Add to function `UnsubscribeVehicleData` response: 
+
+```xml
+<param name="turnSignal" type="VehicleDataResult" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+##### Add to function `GetVehicleData` request: 
+
+```xml
+<param name="turnSignal" type="Boolean" mandatory="false" since="5.0">
+	<description>See TurnSignalData</description>
+	<history>
+		<description>See TurnSignal</description>
+	</history>
+</param>
+```
+
+##### Add to function `GetVehicleData` response: 
+
+```xml
+<param name="turnSignal" type="TurnSignalData" mandatory="false" since="X.x">
+	<description>See TurnSignalData</description>
+	<history>
+		<param name="turnSignal" type="TurnSignal" mandatory="false" since="5.0" until="X.x">
+			<description>See TurnSignal</description>
+		</param>
+	</history>
+</param>
+```
+
+##### Add to function `OnVehicleData` response: 
+
+```xml
+<param name="turnSignal" type="TurnSignalData" mandatory="false" since="X.x">
+	<description>See TurnSignalData</description>
+	<history>
+		<param name="turnSignal" type="TurnSignal" mandatory="false" since="5.0" until="X.x">
+			<description>See TurnSignal</description>
+		</param>
+	</history>
+</param>
+```
+
+#### Updates in HMI_API:
+
+##### Add new struct enum `TurnSignalData` in `Common` interface: 
+
+```xml
+<struct name="TurnSignalData">
+	<param name="turnSignal" type="Common.TurnSignal" mandatory="false" since="5.0">
+		<description>See TurnSignal</description>
+	</param>	
+	<param name="hazardLights" type="Common.VehicleDataStatus" mandatory="false" since="X.x">
+		<description>Status of hazard lights</description>
+	</param>
+</struct>
+
+```
+
+##### Update function `SubscribeVehicleData` request: 
```

Comment:
Can you please remove references to the HMI_API that are not changed by this proposal? It will help to make this proposal more concise and readable for SDLC members.

Reply:
I have only included the changes already, is there any specific reference you are pointing to.

Reply:
With `TurnSignal` being removed, this comment is no longer valid.  Thanks!